From eed6a1083cf5a44d0a7476980ef3471ee2cfd49a Mon Sep 17 00:00:00 2001
Message-ID: <eed6a1083cf5a44d0a7476980ef3471ee2cfd49a.1705743627.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 22:59:01 +0000
Subject: [PATCH 08/38] gnu: go-github-com-aead-chacha20: Move to (gnu packages
 golang-crypto).

* gnu/packages/golang.scm (go-github-com-aead-chacha20): Move from here ...
* gnu/packages/golang-crypto.scm: ... to here.

Change-Id: I97254ae17ba12327649d26f154396c42ed07b716
---
 gnu/packages/golang-crypto.scm | 27 +++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 28 ----------------------------
 2 files changed, 27 insertions(+), 28 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index b2c9c98b44..4462eb6141 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -38,6 +38,33 @@ (define-module (gnu packages golang-crypto)
 ;;;
 ;;; Code:
 
+(define-public go-github-com-aead-chacha20
+  (let ((commit "8b13a72661dae6e9e5dea04f344f0dc95ea29547")
+        (revision "0"))
+    (package
+      (name "go-github-com-aead-chacha20")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/aead/chacha20")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0gbmgq5kbqmbyrsav57ql4jzbvqvp1q7yvcd5fl3wf5g94iyv56r"))))
+      (build-system go-build-system)
+      (arguments
+       `(#:import-path "github.com/aead/chacha20"))
+      (propagated-inputs
+       (list go-golang-org-x-sys))
+      (home-page "https://github.com/aead/chacha20")
+      (synopsis "ChaCha20 and XChaCha20 stream ciphers")
+      (description "ChaCha is a stream cipher family created by Daniel
+Bernstein.  The most common ChaCha variant is ChaCha20 (20 rounds).  ChaCha20
+is standardized in RFC 7539.")
+      (license license:expat))))
+
 (define-public go-github-com-libp2p-go-libp2p-crypto
   (let ((commit "7240b40a3ddc47c4d17c15baabcbe45e5219171b")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 3abb4a2918..358edfed63 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -1832,34 +1832,6 @@ (define-public go-github-com-savsgio-gotils
        "Golang utilities to make your life easier with zero allocations.")
       (license license:asl2.0))))
 
-(define-public go-github-com-aead-chacha20
-  (let ((commit "8b13a72661dae6e9e5dea04f344f0dc95ea29547")
-        (revision "0"))
-    (package
-      (name "go-github-com-aead-chacha20")
-      (version (git-version "0.0.0" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri
-          (git-reference
-           (url "https://github.com/aead/chacha20")
-           (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32 "0gbmgq5kbqmbyrsav57ql4jzbvqvp1q7yvcd5fl3wf5g94iyv56r"))))
-      (build-system go-build-system)
-      (arguments
-       `(#:import-path "github.com/aead/chacha20"))
-      (propagated-inputs
-       (list go-golang-org-x-sys))
-      (home-page "https://github.com/aead/chacha20")
-      (synopsis "ChaCha20 and XChaCha20 stream ciphers")
-      (description "ChaCha is a stream cipher family created by Daniel Bernstein.
-The most common ChaCha variant is ChaCha20 (20 rounds).  ChaCha20 is
-standardized in RFC 7539.")
-      (license license:expat))))
-
 (define-public go-github-com-mufti1-interconv
   (let ((commit "d7c72925c6568d60d361757bb9f2d252dcca745c")
         (revision "0"))
-- 
2.41.0

